### PR TITLE
Add rule: property-blacklist

### PR DESCRIPTION
**Pull request by jeddy3**

### CODE REVIEW

File: src/rules/function-blacklist/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    const validOptions = validateOptions(result, ruleName, {
+      actual: blacklist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+    root.walkDecls(decl => {
+      const { value } = decl
+      valueParser(value).walk(function (node) {
+        if (node.type === "function" && blacklist.indexOf(vendor.unprefixed(node.value)) !== -1) {
+          const charsBeforeColon = decl.toString().indexOf(":")
```
Comment (jeddy3) [inline src/rules/function-blacklist/index.js]: @davidtheclark I could do with some advice here. Is there a better way of working out the index?

File: src/rules/function-blacklist/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          const charsAfterColon = decl.raw("between").length - decl.raw("between").indexOf(":")
+          report({
+            message: messages.rejected(node.value),
+            node: decl,
+            index: charsBeforeColon + charsAfterColon + node.sourceIndex,
```
Comment (jeddy3) [inline src/rules/function-blacklist/index.js]: I could do with some advice here. Is there a better way of working out the index?

File: src/rules/function-blacklist/__tests__/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const testRule = ruleTester(rule, ruleName)
+
+testRule([
+  "rgba",
+  "scale",
+  "linear-gradient",
+], tr => {
+  warningFreeBasics(tr)
+  tr.ok("a { color: pink; }")
+
```
Comment (davidtheclark) [inline src/rules/function-blacklist/__tests__/index.js]: Can you throw in a positive test with a function that is *not* blacklisted?

Comment (jeddy3) [inline src/rules/function-blacklist/__tests__/index.js]: Aww yes, that would be helpful. I got a bit distracted with the the whole index thing...

File: src/rules/function-blacklist/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      actual: blacklist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+    root.walkDecls(decl => {
+      const { value } = decl
+      valueParser(value).walk(function (node) {
+        if (node.type === "function" && blacklist.indexOf(vendor.unprefixed(node.value)) !== -1) {
+          const charsBeforeColon = decl.toString().indexOf(":")
+          const charsAfterColon = decl.raw("between").length - decl.raw("between").indexOf(":")
```
Comment (davidtheclark) [inline src/rules/function-blacklist/index.js]: I don't understand the subtraction here. Why not just `charsBeforeColon + decl.raw("between").length`?

File: src/rules/function-blacklist/__tests__/index.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  tr.notOk("a { color: rgba(0, 0, 0, 0) }", {
+    message: messages.rejected("rgba"),
+    line: 1,
+    column: 12,
+  })
+  tr.notOk("a { background: red, -moz-linear-gradient(45deg, blue, red); }", {
+    message: messages.rejected("-moz-linear-gradient"),
+    line: 1,
+    column: 22,
+  })
```
Comment (davidtheclark) [inline src/rules/function-blacklist/__tests__/index.js]: Would be worth throwing in a test of a nested function (function within function, I mean), and another of a function within a media query.

File: src/rules/function-blacklist/index.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: blacklist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+    root.walkDecls(decl => {
+      const { value } = decl
+      valueParser(value).walk(function (node) {
+        if (node.type === "function" && blacklist.indexOf(vendor.unprefixed(node.value)) !== -1) {
```
Comment (davidtheclark) [inline src/rules/function-blacklist/index.js]: Good call! I'm glad this `unprefixed()` function keeps working for us!

**Comment by davidtheclark** [comment]
Nice job @jeddy3. I just asked for one bit of clarification and a couple of tests.

File: src/rules/function-blacklist/index.js (hunk 4 above)
Comment (jeddy3) [inline src/rules/function-blacklist/index.js]: I originally had it that way, but `decl.raw("between")` returns the characters _before_ and after the colon. So, given

```css
a { transform : scale(1); }
```

`charsBeforeColon + decl.raw("between").length` would trip up by one place. Does that sound about right to you?

I know it looks messy. There has to be a neater way of get the index of `node.value` within a declaration.

**Comment by jeddy3** [comment]
@davidtheclark Thanks for the feedback. I've added all the requested test.

Hopefully we can come up with neater way of get the index of `node.value` within a declaration.

**Pull request merged**